execution_engine_testing/test_support/src/test_context.rs

```diff
@@ -120,6 +120,49 @@ impl TestContext {
             .map_err(Error::from)
     }
 
+    /// Queries for a [`Value`] stored in a dictionary under a given 'name'
+    ///
+    /// Returns an [`Error`] if not found.
+    pub fn query_dictionary_item(
+        &self,
+        key: Key,
+        dictionary_name: Option<String>,
+        dictionary_item_key: String,
+    ) -> Result<Value> {
+        let empty_path = vec![];
+        let dictionary_key_bytes = dictionary_item_key
+            .to_bytes()
+            .map_err(|_| Error::from("Could not serialize key bytes".to_string()))?;
+        let address = match key {
+            Key::Account(_) | Key::Hash(_) => {
+                if let Some(name) = dictionary_name {
+                    let path = vec![name];
+                    let seed_uref = self
+                        .inner
+                        .query(None, key, &path)
+                        .map(Value::new)
+                        .map_err(Error::from)?
+                        .into_t::<URef>()
+                        .map_err(Error::from)?;
+                    Key::dictionary(seed_uref, &dictionary_key_bytes)
+                } else {
+                    return Err(Error::from("No dictionary name was provided".to_string()));
+                }
```

Comment:
If providing a `None` here always returns an error then it would be better to simply expect a non-optional argument.